controllers/platformnetwork_controller.go

```diff
@@ -661,38 +741,261 @@ func (r *PlatformNetworkReconciler) statusUpdateRequired(instance *starlingxv1.P
 	return !instance.Status.DeepEqual(status)
 }
 
-// ReconcileResource interacts with the system API in order to reconcile the
-// state of a data network with the state stored in the k8s database.
-func (r *PlatformNetworkReconciler) ReconcileResource(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (err error) {
-	oldStatus := instance.Status.DeepCopy()
+// Determine if the system type is All-in-one simplex, return the Host object if so.
+func (r *PlatformNetworkReconciler) AIOSXHost(client *gophercloud.ServiceClient) (*hosts.Host, bool, error) {
+	all_hosts, err := hosts.ListHosts(client)
+	if err != nil {
+		return nil, false, common.NewSystemDependency("Failed to get host list")
+	}
 
-	if instance.DeletionTimestamp.IsZero() {
-		err = r.ReconcileAddressPool(client, instance)
-		if err == nil {
-			if networkResourceRequired(instance) {
-				err = r.ReconcileNetwork(client, instance)
+	if len(all_hosts) == 1 {
+		for _, host := range all_hosts {
+			if host.Personality == cloudManager.PersonalityController {
+				default_system, err := system.GetDefaultSystem(client)
+				if err != nil {
+					return nil, false, common.NewSystemDependency("Failed to get default system")
+				}
+
+				if default_system.SystemMode == "simplex" && strings.ToLower(default_system.SystemType) == "all-in-one" {
+					return &host, true, nil
+				}
 			}
 		}
+	}
 
-		inSync := err == nil
+	return nil, false, nil
+}
 
-		if inSync != instance.Status.InSync {
-			r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "synchronization has changed to: %t", inSync)
+func (r *PlatformNetworkReconciler) AddressPoolUpdateRequired(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (bool, error) {
+	var pool *addresspools.AddressPool
+	var err error
+	pool, err = r.FindExistingAddressPool(client, instance)
+	if err != nil {
+		return false, err
+	} else if pool == nil {
+		return true, nil
+	}
+
+	if _, ok := poolUpdateRequired(instance, pool, r); ok {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (r *PlatformNetworkReconciler) NetworkUpdateRequired(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (bool, error) {
+	if instance.Spec.Type == cloudManager.OAMNetworkType {
+		oamNetworkList, err := oamNetworks.ListNetworks(client)
+		if err != nil {
+			return false, err
+		} else if len(oamNetworkList) == 0 {
+			return true, nil
+		} else {
+			oamNetwork := &oamNetworkList[0]
+			if _, ok := oamUpdateRequired(instance, oamNetwork, r); ok {
+				return true, nil
+			}
+		}
+	} else {
+		network, err := r.FindExistingNetwork(client, instance)
+		if err != nil {
+			return false, err
+		} else if network == nil {
+			return true, nil
```

Comment:
what does 'true' indicate here? update required? what will be updated?

Reply:
Returning `true` indicates applied config doesn't match with running config and update is required.